hibernate-core/src/main/java/org/hibernate/type/descriptor/java/JavaTypedExpressible.java

```diff
@@ -11,4 +11,9 @@
  */
 public interface JavaTypedExpressible<T> {
 	JavaType<T> getExpressibleJavaType();
+
+	default int extractHashCode(T disassembledValue) {
```

Comment:
I think this is unnecessary. We should just call the `hashCode` method on the disassembled state directly.